parking_permits_app/models.py

```diff
@@ -0,0 +1,386 @@
+        related_name="customers_other_address",
+        blank=True,
+        null=True,
+    )
+    email = models.CharField(_("Email"), max_length=128, blank=True, null=True)
+    phone_number = models.CharField(
+        _("Phone number"), max_length=32, blank=True, null=True
+    )
+    parking_zone = models.ForeignKey(
+        "ParkingZone", verbose_name=_("Parking zone"), on_delete=models.PROTECT
+    )
+    terms_of_use_accepted = models.BooleanField(null=False, default=False)
+
+    class Meta:
+        db_table = "customer"
+        verbose_name = _("Customer")
+        verbose_name_plural = _("Customers")
+
+    def __str__(self):
+        return "%s - %s %s" % (self.id, self.first_name, self.last_name)
+
+
+class ParkingZone(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Name"), max_length=128, blank=False, null=False)
+    location = models.MultiPolygonField(
+        _("Area (2D)"), srid=settings.SRID, blank=False, null=False
+    )
+
+    class Meta:
+        db_table = "parking_zone"
+        verbose_name = _("Parking zone")
+        verbose_name_plural = _("Parking zones")
+
+    def __str__(self):
+        return "%s - %s" % (self.id, self.name)
+
+
+class VehicleType(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    name = models.CharField(_("Name"), max_length=32, blank=False, null=False)
+
+    class Meta:
+        db_table = "vehicle_type"
+        verbose_name = _("Vehicle type")
+        verbose_name_plural = _("Vehicle types")
+
+    def __str__(self):
+        return "%s - %s" % (self.id, self.name)
+
+
+class LowEmissionCriteria(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    vehicle_type = models.ForeignKey(
+        VehicleType,
+        verbose_name=_("Vehicle type"),
+        on_delete=models.PROTECT,
+        blank=False,
+        null=False,
+    )
+    emission_limit_nedc = models.IntegerField(
+        _("Emission limit (NEDC)"), blank=True, null=True
+    )
+    emission_limit_wltp = models.IntegerField(
+        _("Emission limit (WLTP)"), blank=True, null=True
+    )
+
+    class Meta:
+        db_table = "low_emission_criteria"
+        verbose_name = _("Low-emission criteria")
+        verbose_name_plural = _("Low-emission criterias")
+
+    def __str__(self):
+        return "%s - %s, NEDC: %s, WLTP: %s" % (
+            self.id,
+            self.vehicle_type,
+            self.emission_limit_nedc,
+            self.emission_limit_wltp,
+        )
+
+
+class Address(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    street_name = models.CharField(
+        _("Street name"), max_length=128, blank=False, null=False
+    )
+    street_number = models.CharField(
```

Comment:
For `street_number`, shouldn't we be using `IntegerField`? Is the `CharField` an intentional decision here?

Reply:
This is intentional, because in Finland we have complex scenarios, like `12`, `12 A`, `12a`, `12a-b`and so on. This field would cover all those scenarios, if we define it as char. And this is how it's typically handled in Finland easily.

Reply:
Thanks for the explanation 👍🏼